src/Servers/HttpSys/test/FunctionalTests/RequestBodyTests.cs

```diff
@@ -57,6 +79,29 @@ public async Task RequestBody_ReadAsync_Success()
         }
     }
 
+    [ConditionalFact]
+    public async Task RequestBody_Read0ByteAsync_Success()
+    {
+        string address;
+        using (Utilities.CreateHttpServer(out address, async httpContext =>
+        {
+            Assert.True(httpContext.Request.CanHaveBody());
+            byte[] input = new byte[100];
+            await Task.Delay(1000);
+            int read = await httpContext.Request.Body.ReadAsync(input, 0, 1);
```

Comment:
This initial read in the async test isn't in the sync test. Is that intentional?